### ISSUE TITLE
Focus ring for Shell buttons should be dotted (and slightly inset) to match GTK style

### ISSUE THREAD

**Issue by ya-d** [ubuntu/yaru#749]
Is it intended or a technical limitation that Shell buttons (e.g. shutdown menu) use solid lines for keyboard focus? GTK buttons use doted lines. Also it should be slightly inset to match GTK styling.

<img width="564" alt="focus ring shell" src="https://user-images.githubusercontent.com/30781876/44689144-cf804700-aa56-11e8-9d7b-64086266fb89.png"> <img width="159" alt="focus ring gtk" src="https://user-images.githubusercontent.com/30781876/44689154-d4dd9180-aa56-11e8-89fe-8bd9396635e1.png">

**Comment by Feichtmeier** [comment]
Currently this is made with a box-shadow (we just tool this from upstream)

I played around with this dialogue a lot, since I am not a big fan of the little gap between the edge and the button :) 
We tried the remove the border from the dialogue (I fear I lost the mock-ups but there must be a PR somewhere...) but the orange someone looked very strange at the edges. So we trashed this for good.

The dotted thing could be possible though, I am not sure if. But I will try (my idea would be to try the upstream undershots, they are made with a box-shadow and a gradient I think :thinking: )

**Comment by ya-d** [comment]
That sounds like a lot of hacking… 😅

**Comment by madsrh** [comment]
Quick and dirty mockups:
![image](https://user-images.githubusercontent.com/3986894/44706086-c4520900-aaa0-11e8-85b5-f10736cd3cb7.png)

![image](https://user-images.githubusercontent.com/3986894/44706097-cb791700-aaa0-11e8-9be6-89e6c969f75b.png)

> my idea would be to try it like the upstream undershots, they are made with a box-shadow and a gradient I think

I'm keen to see what that looks like, but I'm not sure it solves the issue here which was that focus rings should be styled alike.

**Comment by Feichtmeier** [comment]
Sorry if I was not clear enough:

Currently this is made with inset box shadow
Box shadows can't be simply dashed line borders
For some reason borders look strange in this dialogue
So if we want dashed we need to trick the box Shadow
The upstream undershoots are also some kind of dashed box shadow trick which I could maybe steal to use it here

**Comment by Feichtmeier** [comment]
Oh oh... bad news. The trick does not work (it was from gtk so I wasn't sure about it anyways)
And the shell does not like dashed borders (it does not work)

I guess we close this then due to technical limitations

**Comment by Feichtmeier** [comment]
But talking about the dialogues, how about making the buttons flat?
This would reduce the awkwardness of the selection ring above pressed buttons
![peek 2018-08-28 11-54](https://user-images.githubusercontent.com/15329494/44715892-36cee300-aab9-11e8-8800-27682d866742.gif)
![peek 2018-08-28 11-55](https://user-images.githubusercontent.com/15329494/44715912-46e6c280-aab9-11e8-802c-a23596b2d794.gif)

**Comment by clobrano** [comment]
It looks good to me :+1:

**Comment by madsrh** [comment]
You really don't like that 3d look 😉 LOL

I'm fine with this proposal 👍

**Comment by Feichtmeier** [comment]
I would be okay with it if it would not look so unfinished (subjective). That's why I prefer this clear and sharp solution. Anyways, we can test it on a PR =) (I made a post on snapcraft about the failing builds ... )

**Comment by Feichtmeier** [comment]
Closing this initial issue, since we can not fix it yet with the current shell technology
